File: vanilla-mvp/src/main/java/be/vergauwen/simon/cleanmvp/ui/TestContract.java
package be.vergauwen.simon.cleanmvp.ui;

import be.vergauwen.simon.himura.MVPContract;

interface TestContract {
    interface View extends MVPContract.View {
        void showThings();
    }

    interface Presenter extends MVPContract.Presenter<View> {
        void loadThings();
    }
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/ui/TestContract.kt
package be.vergauwen.simon.kotlindaggermvp.ui

import be.vergauwen.simon.himurakotlin.MVPContract

interface TestContract {
  interface View : MVPContract.View {
    fun showThings()
  }

  interface Presenter<V : View> : MVPContract.Presenter<V> {
    fun loadThings()
  }

  interface Component<V : View ,P : Presenter<V>> : MVPContract.Component<V,P>
}